### PR TITLE
Make replace_rules_internal return true only if important rules changed.

### PR DESCRIPTION
**Pull request by hiikezoe**
https://bugzilla.mozilla.org/show_bug.cgi?id=1367975

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17951)

### CODE REVIEW

**Comment by hiikezoe** [comment]
Waiting for a final try.
https://treeherder.mozilla.org/#/jobs?repo=try&revision=690abc698749583b343d34cf99d0b0b9b4d42e6c

File: components/style/rule_tree/mod.rs [hunk 1]
```diff
@@ -335,6 +337,9 @@ impl RuleTree {
         // special cases, and replacing them for a `while` loop, avoiding the
         // optimizations).
         if current.get().level == level {
+            if level.is_important() {
+                *important_rules_changed = true;
```
Comment (emilio) [inline components/style/rule_tree/mod.rs]: Maybe just:

```
*important_rules_changed |= level.is_importan();
```

But this is very minor :)

Comment (hiikezoe) [inline components/style/rule_tree/mod.rs]: oh right. thanks.

**Pull request merged**